libs/sdk-core/src/persist/transactions.rs

```diff
@@ -8,6 +8,9 @@ use rusqlite::{named_params, params, OptionalExtension};
 use std::collections::HashSet;
 
 use std::str::FromStr;
+use serde_json::{Value, Map};
+
+const METADATA_MAX_SIZE_KB: usize = 10000;
```

Comment:
No, it's actually kilobytes. Now that I think about it though, it probably is too much. Let's set a max of 10kb, and see how it goes, since the string encoding is UTF-8 and may be heavier for some locales (up to 4 bytes per character, meaning 2500 characters)

Reply:
👍 I thought 10MB might be too much 😄